rotkehlchen/chain/evm/contracts.py

```diff
@@ -131,6 +135,115 @@ def decode_event(
         return decode_event_data_abi(tx_log=tx_log, event_abi=event_abi)  # type: ignore
 
 
+def try_download_abi_from_remote(name: str) -> Optional[list[dict[str, Any]]]:
+    """
+    Checks whether abi with the given name exists on remote.
+    If it does, downloads it and adds to globaldb.
+
+    Returns the abi if found and None otherwise.
+    """
+    abi_url = f'{BASE_CONTRACT_DATA_URL}/abi/{name}.json'
+    try:
+        abi_data = request_get(abi_url)
+    except (RemoteError, UnableToDecryptRemoteData) as e:
+        log.error(f'Could not find abi {name} in rotki data repo due to: {str(e)}')
+        return None
+
+    if isinstance(abi_data, list) is False:
+        log.error(f'Abi should be a list. But we got {abi_data}')
+        return None
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        write_cursor.execute(
+            'INSERT INTO contract_abi(name, value) VALUES(?, ?)',
+            (name, json.dumps(abi_data, separators=(',', ':'))),
+        )
+
+    return abi_data  # type: ignore[return-value]  # abi_data is a list here for sure
+
+
+def try_download_contract_from_remote(name: str, target_chain_id: ChainID) -> Optional[EvmContract]:  # noqa: E501
+    """
+    Checks whether contract with the given name exists on remote.
+    If it does, downloads it and adds to globaldb.
+    If abi of the contract already exists in the globaldb, reuses that abi, otherwise
+    adds a new abi.
+
+    Returns the contract if found and None otherwise.
+    """
+    contract_url = f'{BASE_CONTRACT_DATA_URL}/contract/{name}.json'
+    try:
+        contract_data = request_get_dict(contract_url)
+    except (RemoteError, UnableToDecryptRemoteData) as e:
+        log.error(f'Could not find contract {name} in rotki data repo due to: {str(e)}')
+        return None
+
+    if (
+        'address' not in contract_data or
+        'chain_id' not in contract_data or
+        'abi' not in contract_data or
+        'abi_name' not in contract_data or
+        'deployed_block' not in contract_data
+    ):
+        log.error(
+            f'Expected to see address, chain_id, abi and deployed_block in contract_data but have '
+            f'not found them. The contract data was: {contract_data}',
+        )
+        return None
+
+    if contract_data['chain_id'] != target_chain_id.value:
+        log.error(
+            f'Expected contract {name} to have chain id {target_chain_id.value}, '
+            f'but it had chain id {contract_data["chain_id"]}',
+        )
+        return None
```

Comment:
what about contracts deployed in different chains with the same name? should maybe `chain_id` be a list

Reply:
Good question.

> should maybe chain_id be a list

I wouldn't do this since `try_download_contract_from_remote` is always called for a particular chain id.

However I think it will make sense in the contract url/filename. So that a contract url is computed from name + chain id. What do you think?

Reply:
Then we would need to mantain duplicated information right? or do you have something in mind to avoid that?

Reply:
No-no, where will the duplicate be? We can just have the chain id in the url. No need to store it in the json data. Or am I missing something?